.github/workflows/platform-release.yaml

```diff
@@ -194,33 +226,108 @@ jobs:
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          repository: jozu-ai/homebrew-kitops
+          ref: 'main'
+          path: homebrew-kitops
+          token: ${{ secrets.GITOPS_PAT }}
+            
+      - name: Download the homebrew artifacts to build the homebrew recipe
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.3
+        with:
+          name: homebrew-files
+          path: homebrew-files
+          merge-multiple: true
+
+      - name: Create and save homebrew-metadata.txt
+        env:
+          TAG_NAME: ${{ inputs.release_tag}}
+          REPO: ${{ github.repository }}
+        run: |
+          shopt -s failglob
+          pushd homebrew-files
+          awk '
+            BEGIN { tag_name = ENVIRON["TAG_NAME"]; repo = ENVIRON["REPO"];}
+            /.tar.gz$/ { 
+             print "\""$1"\"", 
+              "\"https://github.com/" repo "/releases/download/" tag_name "/" $2 "\"", 
+              $2, 
+              tag_name }' kitops_${TAG_NAME}_checksums.txt |
+          awk '{ sub(/.tar.gz/, "", $3); print $1, $2, $3, $4 }' |
+          awk '{ sub(/kitops-/, "", $3); print $1, $2, $3, $4 }' |
+          awk '{sub(/v/, "", $4); print $1, $2, $3, $4 }' > homebrew-metadata.txt
+          popd
+
+      # uses the homebrew-metadata.txt file generated in the
+      # previous step to generate the homebrew formula (kitops.rb) 
+      # for the homebrew tap (located at 'jozu-ai/homebrew-kitops')
+      - name: Create homebrew formula
+        run: |
+          pushd homebrew-files
+          awk -f create-homebrew-recipe.awk homebrew-metadata.txt
+          popd
+
+      # publish the homebrew recipe file to the homebrew tap
+      # located in the repository: 'jozu-ai/homebrew-kitops'
+      - name: Commit homebrew recipe to the homebrew tap
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITOPS_PAT }}
+          TAG_NAME: ${{ inputs.release_tag}}
+          REPO: "jozu-ai/homebrew-kitops"
+        run: |
+          cp -f ./homebrew-files/homebrew-metadata.txt ./homebrew-kitops/.
+          cp -f ./homebrew-files/kitops.rb ./homebrew-kitops/.
+          pushd homebrew-kitops
+          CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+          PR_BRANCH="${{ github.ref_name }}-homebrew-tap-update"
+          git fetch origin main
+          git branch "$PR_BRANCH"
+          git checkout "$PR_BRANCH"
+          git pull origin --ff-only "${PR_BRANCH}" || true
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
+          git add --all
+          git commit -m "homebrew: update Homebrew Tap Formula for ${{ github.ref_name }}"
+          git push origin "${PR_BRANCH}"
+          gh pr create --fill --base main --head "${PR_BRANCH}"
+          git checkout "${CURRENT_BRANCH}"
+          popd
```

Comment:
This could potentially be simpler using [peter-evans/create-pull-request](https://github.com/peter-evans/create-pull-request) -- IIRC we use this elsewhere